plotting.py

```diff
@@ -1063,17 +1063,18 @@ def _check_consistency(self):
                 self.fields = list(series_names)
             else:
                 raise InputError('attempting to plot multiple series with different field names')
-        # If fields='all', retrieve fields from dataset
-        elif self.fields=='all':
-            self.fields = _get_fieldnames(list(self.datasets.values())[0])
-            assert(all([_get_fieldnames(df)==self.fields for df in self.datasets.values()])), \
-               "The option fields = 'all' only works when all datasets have the same fields"
-            # No fieldnames, so use generic field name
-            if self.fields == [None,]:
-                self.fields = ['field',]
-        # If fields is a single instance, convert to a list
         elif isinstance(self.fields,str):
-            self.fields = [self.fields,]
+            # If fields='all', retrieve fields from dataset
+            if self.fields=='all':
+                self.fields = _get_fieldnames(list(self.datasets.values())[0])
+                assert(all([_get_fieldnames(df)==self.fields for df in self.datasets.values()])), \
+                   "The option fields = 'all' only works when all datasets have the same fields"
+                # No fieldnames, so use generic field name
+                if self.fields == [None,]:
+                    self.fields = ['field',]
```

Comment:
What's the purpose of  requesting a field called "field"? Is this for the series plots?

Reply:
Yes, a Pandas series does not necessarily have a name so if no fieldname is found, just give it a generic name. I'm not sure whether it would work leaving the fieldname equal to None.

Reply:
Could you update the "use generic field name" comment to mention usage with series, and then rename 'field' to something slightly more obvious of your intent, e.g., `self.fields = ['unnamed_series',]`?

Reply:
I further tested the plotting library with series and it turned out that the generic field name did not work appropriately. I changed it so that now, if the series does not have a name and no fields are specified, the list of fields is set to [None] which is compatible with the list of available fields, which also returns [None] for a series without a name. Can you check and let me know if this behavior needs more documentation?

Reply:
I think this is fine.